src/librustc/hir/mod.rs

```diff
@@ -2289,6 +2289,7 @@ pub struct CodegenFnAttrs {
     pub flags: CodegenFnAttrFlags,
     pub inline: InlineAttr,
     pub export_name: Option<Symbol>,
+    pub link_name: Option<Symbol>,
```

Comment:
`link_name` and `export_name` are kind of the same and should never occur at the same time, right? Could you add a comment on why there are two separate fields for it?

Reply:
Sure thing, docs added!